core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -584,18 +583,17 @@ private SqlNode makCastCallForTimeWithPrecision(SqlNode operandToCast, int preci
   private void unparseRegexSubstr(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
     SqlCall extractCall;
     switch (call.operandCount()) {
-    case 3:
+    case 2: case 3:
```

Comment:
Can we remove the switch case as both 2,3 and 4,5 operand count have same code now.

Reply:
I have added the changes as per the review comments as I observed no offset is been added at bigquery end as per examples provide at bigquery side.